src/rbac/roleManager.ts

```diff
@@ -34,3 +38,27 @@ export interface RoleManager {
   // PrintRoles prints all the roles to log.
   printRoles(): Promise<void>;
 }
+
+export interface SyncedRoleManager {
```

Comment:
we should add an optional method in RoleManager to distinguish whether it is synchronous or asynchronous, so like:
```
synced?(): boolean;
```